src/slskd/Application.cs

```diff
@@ -941,7 +943,7 @@ private async Task<SearchResponse> SearchResponseResolver(string username, int t
                         Client.Username,
                         token,
                         uploadSpeed: State.CurrentValue.User.Statistics.AverageSpeed,
-                        freeUploadSlots: forecastedPosition == 0 ? 1 : 0,
+                        freeUploadSlots: group.Slots - group.UsedSlots,
```

Comment:
This doesn't look right. It's supposed to be a boolean value indicating if there are free slots.

Reply:
I'm wondering what I saw that lead me to believe this.  I'm reading a byte, but there's also some logic after that to conditionally read a 32 or 64 bit integer: https://github.com/jpdillingham/Soulseek.NET/blob/master/src/Messaging/Messages/Peer/SearchResponseFactory.cs#L46-L64

Reply:
> there's also some logic after that to conditionally read a 32 or 64 bit integer: 

I looked this up some months ago, and came to the conclusion that the assumption about a single 64 bit integer was incorrect (the official clients only displayed a maximum value of the 32 bit integer limit for the queue length): https://github.com/nicotine-plus/nicotine-plus/commit/71a92c061971a89412da7004d3828e891f3be965

Reply:
Thanks for the heads up!